core/query.go

```diff
@@ -0,0 +1,10 @@
+package core
+
+type Query struct {
+	Context QueryContext
+}
+
+type QueryContext struct {
+	// Group
```

Comment:
side note: @sipsma if we ever want to, we could use this same mechanism for a `query.WithDefaultPlatform()`